idea/idea-frontend-fir/src/org/jetbrains/kotlin/idea/references/FirReferenceResolveHelper.kt

```diff
@@ -369,11 +374,33 @@ internal object FirReferenceResolveHelper {
         val classId = fir.classId ?: return emptyList()
 
         var parent = expression.parent as? KtDotQualifiedExpression
+        val referencedName = expression.getReferencedName()
+        if (parent != null && expression == parent.receiverExpression && classId.packageFqName.startsWith(Name.identifier(referencedName))) {
+            // the expression references a package under the root
+            return listOfNotNull(symbolBuilder.createPackageSymbolIfOneExists(FqName(referencedName)))
+        }
         // Distinguish A.foo() from A(.Companion).foo()
         // Make expression.parent as? KtDotQualifiedExpression local function
         while (parent != null) {
             val selectorExpression = parent.selectorExpression ?: break
             if (selectorExpression === expression) {
+                val selectedFqNameSegments = parent.fqNameSegments()
+                if (selectedFqNameSegments != null) {
```

Comment:
Please, add corresponding tests to `AbstractFirReferenceResolveTest` which checks resolve on nested qualifiers

Reply:
Thanks! Actually while working on this I realized there are several issues (https://youtrack.jetbrains.com/issue/KT-47135, https://youtrack.jetbrains.com/issue/KT-47121, https://youtrack.jetbrains.com/issue/KT-47122) in FIR that's related to this change. So I will move these changes into a separate PR instead.

As a result, two of the tests for ImportMemberAction is currently failing and disabled.